internal/resource/configmap.go

```diff
@@ -90,116 +91,115 @@ func (builder *ServerConfigMapBuilder) Update(object client.Object) error {
+			if _, err := userConfigurationSection.NewKey("web_mqtt.ssl.port", "15676"); err != nil {
 				return err
 			}
-			if _, err := defaultSection.NewKey("web_mqtt.ssl.cacertfile", caCertPath); err != nil {
+			if _, err := userConfigurationSection.NewKey("web_mqtt.ssl.cacertfile", caCertPath); err != nil {
 				return err
 			}
-			if _, err := defaultSection.NewKey("web_mqtt.ssl.certfile", tlsCertPath); err != nil {
+			if _, err := userConfigurationSection.NewKey("web_mqtt.ssl.certfile", tlsCertPath); err != nil {
 				return err
 			}
-			if _, err := defaultSection.NewKey("web_mqtt.ssl.keyfile", tlsKeyPath); err != nil {
+			if _, err := userConfigurationSection.NewKey("web_mqtt.ssl.keyfile", tlsKeyPath); err != nil {
 				return err
 			}
 			if builder.Instance.DisableNonTLSListeners() {
-				if _, err := defaultSection.NewKey("web_mqtt.tcp.listener", "none"); err != nil {
+				if _, err := userConfigurationSection.NewKey("web_mqtt.tcp.listener", "none"); err != nil {
 					return err
 				}
 			}
 		}
 		if builder.Instance.AdditionalPluginEnabled("rabbitmq_web_stomp") {
-			if _, err := defaultSection.NewKey("web_stomp.ssl.port", "15673"); err != nil {
+			if _, err := userConfigurationSection.NewKey("web_stomp.ssl.port", "15673"); err != nil {
 				return err
 			}
-			if _, err := defaultSection.NewKey("web_stomp.ssl.cacertfile", caCertPath); err != nil {
+			if _, err := userConfigurationSection.NewKey("web_stomp.ssl.cacertfile", caCertPath); err != nil {
 				return err
 			}
-			if _, err := defaultSection.NewKey("web_stomp.ssl.certfile", tlsCertPath); err != nil {
+			if _, err := userConfigurationSection.NewKey("web_stomp.ssl.certfile", tlsCertPath); err != nil {
 				return err
 			}
-			if _, err := defaultSection.NewKey("web_stomp.ssl.keyfile", tlsKeyPath); err != nil {
+			if _, err := userConfigurationSection.NewKey("web_stomp.ssl.keyfile", tlsKeyPath); err != nil {
 				return err
 			}
 			if builder.Instance.DisableNonTLSListeners() {
-				if _, err := defaultSection.NewKey("web_stomp.tcp.listener", "none"); err != nil {
+				if _, err := userConfigurationSection.NewKey("web_stomp.tcp.listener", "none"); err != nil {
 					return err
 				}
 			}
 		}
 	}
 
 	if builder.Instance.MemoryLimited() {
-		if _, err := defaultSection.NewKey("total_memory_available_override_value", fmt.Sprintf("%d", removeHeadroom(builder.Instance.Spec.Resources.Limits.Memory().Value()))); err != nil {
+		if _, err := userConfigurationSection.NewKey("total_memory_available_override_value", fmt.Sprintf("%d", removeHeadroom(builder.Instance.Spec.Resources.Limits.Memory().Value()))); err != nil {
 			return err
 		}
 	}
 
-	rmqProperties := builder.Instance.Spec.Rabbitmq
-	if err := cfg.Append([]byte(rmqProperties.AdditionalConfig)); err != nil {
-		return fmt.Errorf("failed to append spec.rabbitmq.additionalConfig: %w", err)
-	}
-
-	var rmqConfBuffer bytes.Buffer
+	// TODO refactor: use string builder
+	var rmqConfBuffer strings.Builder
```

Comment:
nitpick: not a buffer anymore, maybe rename it to a `rmqConfBuilder`?

Reply:
The string builder is sort of a buffer, specialised for strings, and we use the word "Builder" for a different purpose in the `resource` package.

https://go.googlesource.com/go/+/go1.15.6/src/strings/builder.go#15

Reply:
Good!